.repo/sample-list-generator/src/types.ts

```diff
@@ -0,0 +1,36 @@
+import type { AvailableFolderTypes } from './constants';
+
+export interface ApiItem {
+  type: ApiTypeResult;
+  catagory: string;
+  name: string;
+}
+
+export interface ManifestMetadata {
+  title: string;
+  description: string;
+  permissions: string[];
+}
+
+export type SampleItem = {
+  type: 'API_SAMPLE' | 'FUNCTIONAL_SAMPLE';
+  name: string;
+  repo_link: string;
+  apis: ApiItem[];
+} & ManifestMetadata;
+
+export interface AvailableFolderItem {
+  path: string;
+  type: AvailableFolderTypes;
+}
+
+export type ApiTypeResult =
+  | 'event'
+  | 'method'
```

Comment:
Can we use `function`? I know it doesn't have that name elsewhere, but it's more accurate for JS.

Reply:
I noticed that developer.chrome.com uses the term "method(s)". Perhaps using "method(s)" would be easier to understand.🤔

Reply:
I think you're right actually. I was thinking that we should slowly change developer.chrome.com to say functions too, but looking in to it, it does seem like even MDN uses the term "method" for a function on an object property.